.github/generate-doc.py

```diff
@@ -12,6 +14,137 @@ class Colors:
     RESET = '\033[0m'
 
 
+def format_markdown_file(filename):
+    with open(filename, 'r+') as file:
+        content = file.read()
+        formatted_content = markdown2.markdown(content)
+    with open(filename, 'w') as file:
+        file.write(formatted_content)
+    replace_string_in_markdown(filename, "<p>", "")
+    replace_string_in_markdown(filename, "</p>", "")
+    replace_string_in_markdown(filename, "<em>", "_")
+    replace_string_in_markdown(filename, "</em>", "_")
```

Comment:
I'm honestly not a big fan of having to deal with html tags here. Did you consider a library like [mdformat](https://pypi.org/project/mdformat/) instead? The purpose of markdown2 is to be able to display markdown content on websites which we don't really need